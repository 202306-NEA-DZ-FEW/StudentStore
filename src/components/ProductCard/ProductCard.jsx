import Image from "next/image";
import Link from "next/link";
import React from "react";
import { useContext } from "react";
import { FaCartPlus } from "react-icons/fa";

import { CartContext } from "@/context/CartContext";

function ProductCard({ product }) {
    const { addItemToCart } = useContext(CartContext);
    const addProductToCart = () => addItemToCart(product);
    return (
        <div className='card w-[220px]  bg-white shadow-xl'>
            <div className='relative'>
                <Link href=''>
                    <figure className='w-full h-[180px] rounded-t-2xl  relative'>
                        <Image
                            fill
                            className='w-full h-full group-hover:scale-110 duration-200 object-cover'
                            src={product.pictures[1]}
                            alt={product.title}
                            priority
                        />
                    </figure>
                </Link>
                <div className='absolute top-0 m-2  left-0 rounded-full '>
                    <p
                        className={`${
                            product?.type === "sale"
                                ? "bg-[#1B96EF]"
                                : "bg-[#FF8A57]"
                        } rounded-full  p-1 text-[11px] font-bold capitalize tracking-wide text-white sm:py-1 sm:px-3`}
                    >
                        {product?.type}
                    </p>
                </div>
            </div>

            <div className='mt-1 flex flex-col justify-between px-3 p-2  mb-2 '>
                <h2 className=' text-slate-900 text-xl max-w-lg  tracking-tight font-semibold text-center '>
                    {product.title.length > 18
                        ? product.title.substring(0, 18) + "..."
                        : product.title}
                </h2>
                <div className='mt-3 mb-4 w-full flex items-center justify-between text-slate-600'>
                    <p className='badge bg-slate-600 text-white'>
                        {product?.category}
                    </p>
                    <p className='capitalize'>{product?.location.city}</p>
                </div>
                <div className=' flex items-center   justify-between font-semibold text-slate-900 '>
                    <p className='text-xl '>{product?.price}$</p>
                    <button
                        onClick={addProductToCart}
                        className='text-3xl  text-[#4fd65a] hover:text-[#76be7c]'
                    >
                        <FaCartPlus />
                    </button>
                </div>
            </div>
        </div>
    );
}

export default ProductCard;
