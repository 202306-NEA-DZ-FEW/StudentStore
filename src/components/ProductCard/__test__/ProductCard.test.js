import renderer from "react-test-renderer";

import ProductCard from "../ProductCard";

it("renders correctly", () => {
    const tree = renderer.create(<ProductCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
