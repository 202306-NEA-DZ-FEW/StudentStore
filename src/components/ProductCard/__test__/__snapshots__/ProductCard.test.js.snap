// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card w-[270px]  bg-white shadow-xl"
>
  <div
    className="relative flex justify-center items-center m-4 mb-0 rounded-lg h-[80%]  "
  >
    <figure
      className="  rounded-lg "
    >
      <img
        alt="Picture of the product"
        className="object-cover  scale-100 rounded-lg transition-transform hover:scale-105 duration-300 ease-in-out overflow-hidden"
        data-nimg="1"
        decoding="async"
        height={200}
        onError={[Function]}
        onLoad={[Function]}
        src=""
        style={
          Object {
            "color": "transparent",
          }
        }
        width={200}
      />
      <div
        className="absolute top-0 m-2  left-0 rounded-full "
      >
        <p
          className="rounded-full bg-[#7874F2] p-1 text-[8px] font-bold uppercase tracking-wide text-white sm:py-1 sm:px-3"
        >
          Sale
        </p>
      </div>
    </figure>
  </div>
  <div
    className="mt-1 px-5 p-4  mb-2 "
  >
    <h2
      className="card-title text-slate-900 text-xl max-w-lg tracking-tight font-bold text-center "
    >
      Nike Air MX Super 2500
    </h2>
    <div
      className="mt-3 mb-4 w-full  flex items-center   justify-between   text-slate-600 "
    >
      <p
        className=" badge badge-outline "
      >
        category
      </p>
      <p>
        location
      </p>
    </div>
    <div
      className=" flex items-center   justify-between font-semibold text-slate-900 "
    >
      <p
        className="text-xl "
      >
        $449
      </p>
      <button
        className="text-3xl text-[#FF8A57] hover:text-orange-500"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
